### DISCUSSION TITLE
Limited History Messages

### DISCUSSION THREAD

**Discussion by SaltedFish23** [open-webui/open-webui#5941] [General]
**Is your feature request related to a problem? Please describe.**
Sometimes i just ask for one specific problem that do not need the context, and it is a waste to send all the history messages for such problem. And it is hard to switch the context length to achieve the goal of just send the latest problem to api.

**Describe the solution you'd like**
Allow me to switch how many history messages that will be send to api in the chat.

**Describe alternatives you've considered**
Maybe just to include the specific numbers of history messages before send the input and history to endpoint?

**Additional context**
<img width="220" alt="image" src="https://github.com/user-attachments/assets/75e41361-13c1-4a28-b42a-6a02b42a0ccc">

**Comment by rgaricano** [comment]
you can use a filter tool:
https://openwebui.com/f/hub/context_clip_filter
for conv:
https://openwebui.com/f/grayxu/cut_context

**Comment by Cnly** [comment]
Without built-in support, I think one problem with using a function is that you can only set the valve per-user, not per-chat, which makes it much less useful IMO...

**Comment by xirotech** [comment]
I think this is very necessary, in most modern AI clients today, limiting the number of “history chats” sent at a time is a mandatory option, and often the user needs to actively adjust this parameter for different models, because some models tend to ignore things, and if you send it all history chats + new prompts, it will ignore the new prompts and think about the history it has already worked on.

**Comment by jwsky** [comment]
same question,hope to support that it can limit the history chat counts
